src/test/java/uk/gov/hmcts/darts/cases/service/impl/CaseServiceImplTest.java

```diff
@@ -0,0 +1,418 @@
+
+        assertEquals(SWANSEA, savedCaseEntity.getCourthouse().getCourthouseName());
+        assertEquals("2", savedCaseEntity.getCaseNumber());
+        assertNotNull(savedCaseEntity.getDefendants());
+        assertNotNull(savedCaseEntity.getProsecutors());
+        assertNotNull(savedCaseEntity.getDefenders());
+
+
+        assertEquals("1", savedHearingEntity.getCourtroom().getName());
+        assertEquals(SWANSEA, savedHearingEntity.getCourtroom().getCourthouse().getCourthouseName());
+        assertNotNull(savedHearingEntity.getJudges());
+        assertNotNull(savedHearingEntity.getCourtCase());
+
+        String actualResponse = objectMapper.writeValueAsString(result);
+        String expectedResponse = getContentsFromFile(
+            "Tests/cases/CaseServiceTest/addCase/expectedResponseWithCourtroom.json");
+        JSONAssert.assertEquals(expectedResponse, actualResponse, JSONCompareMode.NON_EXTENSIBLE);
+    }
+
+
+    @Test
+    @SuppressWarnings("PMD.JUnitTestsShouldIncludeAssert")
+    void testAddCaseWithoutCourtroom() throws IOException {
+        CourthouseEntity courthouseEntity = CommonTestDataUtil.createCourthouse(SWANSEA);
+
+        Mockito.when(courthouseRepository.findByCourthouseName(any())).thenReturn(Optional.of(courthouseEntity));
+        Mockito.when(caseRepository.saveAndFlush(any())).thenAnswer(invocation -> {
+            Object[] args = invocation.getArguments();
+            return args[0];
+        });
+
+        AddCaseRequest request = CommonTestDataUtil.createAddCaseRequest(null);
+        ScheduledCase result = service.addCaseOrUpdate(request);
+
+        Mockito.verify(caseRepository).saveAndFlush(caseEntityArgumentCaptor.capture());
+        Mockito.verifyNoInteractions(hearingRepository);
+
+        CaseEntity savedCaseEntity = caseEntityArgumentCaptor.getValue();
+
+        assertEquals(SWANSEA, savedCaseEntity.getCourthouse().getCourthouseName());
+        assertEquals("2", savedCaseEntity.getCaseNumber());
+        assertNotNull(savedCaseEntity.getDefendants());
+        assertNotNull(savedCaseEntity.getProsecutors());
+        assertNotNull(savedCaseEntity.getDefenders());
+
+        String actualResponse = objectMapper.writeValueAsString(result);
+        String expectedResponse = getContentsFromFile(
+            "Tests/cases/CaseServiceTest/addCase/expectedResponseWithoutCourtroom.json");
+        JSONAssert.assertEquals(expectedResponse, actualResponse, JSONCompareMode.NON_EXTENSIBLE);
+    }
+
+    @Test
+    void testAddCaseNonExistingCourthouse() {
+
+        AddCaseRequest request = CommonTestDataUtil.createAddCaseRequest(null);
+
+        DartsApiException thrownException = assertThrows(
+            DartsApiException.class,
+            () -> service.addCaseOrUpdate(request)
+        );
+
+        assertEquals("Provided courthouse does not exist", thrownException.getMessage());
+    }
+
+    @Test
+    void testAddCaseNonExistingCourtroom() {
```

Comment:
Another courtroom one which I don't think is correct.

Reply:
The result is wrong as it should automatically create the courtroom as per your previous comment but I think we should keep the test to cover that scenario.